cmd/module.go

```diff
@@ -41,7 +43,10 @@ func init() {
 		Use:   "worker",
 		Short: "gogstash worker mode",
 		RunE: func(ctx context.Context, cmd *cobra.Command, args []string) error {
-			return gogstash(ctx, flagConfig.String(), flagDebug.Bool(), flagPProf.String(), true)
+			err := gogstash(ctx, flagConfig.String(), flagDebug.Bool(), flagPProf.String(), true)
+			sentry.CurrentHub().Recover(err)
+			sentry.Flush(time.Second * 5)
```

Comment:
Did you check the recover function work when the application panic? It's better to write some tests.

Reply:
We tested it, Since Sentry is an external dependency, writing unit tests to directly validate its behavior in case of panics would be challenging.

Reply:
The test still panic.

ref: https://docs.sentry.io/platforms/go/usage/panics/

```go
func TestSentryRecover(t *testing.T) {
	err := sentry.Init(sentry.ClientOptions{
		Dsn:              "",
		TracesSampleRate: 1.0,
	})
	require.NoError(t, err)

	require.NotPanics(t, func() {
		err := func() error {
			panic("panic")
		}()
		sentry.CurrentHub().Recover(err)
		sentry.Flush(time.Second * 5)
	})
}
```